src/services/permissions-cache.js

```diff
@@ -0,0 +1,168 @@
+          newPermissions[modelName].actions = {
+            ...newPermissions[modelName].actions,
+            [actionName]: {
+              triggerEnabled: action.users ? action.allowed && action.users : action.allowed,
+            },
+          };
+        });
+      }
+    });
+
+    return newPermissions;
+  }
+
+  static _setTeamsACLPermissions(renderingId, permissions) {
+    const newFormatPermissions = permissions
+      ? PermissionsCache.transformPermissionsFromOldToNewFormat(permissions)
+      : null;
+
+    PermissionsCache.permissions.renderings = {
+      ...PermissionsCache.permissions.renderings,
+      [renderingId]: {
+        data: newFormatPermissions,
+        lastRetrieve: moment(),
+      },
+    };
+  }
+
+  static _setRolesACLPermissions(renderingId, permissions) {
+    PermissionsCache.permissions.collections = {
+      data: permissions.collections,
+      lastRetrieve: moment(),
+    };
+    PermissionsCache.permissions.renderings = {
+      ...PermissionsCache.permissions.renderings,
+      [renderingId]: {
+        data: permissions.renderings ? permissions.renderings[renderingId] : null,
+        lastRetrieve: moment(),
+      },
+    };
+  }
+
+  static _setPermissions(renderingId, permissions) {
+    if (PermissionsCache.isRolesACLActivated) {
+      PermissionsCache._setRolesACLPermissions(renderingId, permissions);
+    } else {
+      PermissionsCache._setTeamsACLPermissions(renderingId, permissions);
+    }
+  }
+
+  static getLastRetrieveTime(renderingId, permissionName) {
+    // In the case of rolesACL format and browseEnabled permission, the last retrieve to be taken
+    // into account is the one stored by rendering (because of the scope information).
+    if (PermissionsCache.isRolesACLActivated && PermissionsCache._getCollectionsPermissions() && permissionName !== 'browseEnabled') {
+      return PermissionsCache._getCollectionsPermissions().lastRetrieve;
+    }
+    return PermissionsCache._getRenderingPermissions(renderingId)
+      ? PermissionsCache._getRenderingPermissions(renderingId).lastRetrieve
+      : null;
+  }
+
+  static resetExpiration(renderingId) {
+    if (PermissionsCache.isRolesACLActivated && PermissionsCache._getCollectionsPermissions()) {
+      PermissionsCache._getCollectionsPermissions().lastRetrieve = null;
+    }
+
+    if (PermissionsCache._getRenderingPermissions(renderingId)) {
+      PermissionsCache._getRenderingPermissions(renderingId).lastRetrieve = null;
+    }
+  }
+
+  static isPermissionExpired(renderingId, permissionName) {
+    const currentTime = moment();
+    const lastRetrieve = PermissionsCache.getLastRetrieveTime(renderingId, permissionName);
+
+    if (!lastRetrieve) {
+      return true;
+    }
+
+    const elapsedSeconds = currentTime.diff(lastRetrieve, 'seconds');
+    return elapsedSeconds >= PermissionsCache.expirationInSeconds;
+  }
+
+  static async retrievePermissions(environmentSecret, renderingId) {
```

Comment:
As the function to retrieve the permissions is here plus there is also all the function. to get the permissions I think this service should be called `PermissionsGetter`